src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -704,6 +702,42 @@ public async Task TestSqlSettingPostStartupConfigurations(string configurationEn
             Assert.AreEqual(HttpStatusCode.BadRequest, postConfigOpenApiSwaggerEndpointAvailability.StatusCode);
         }
 
+        /// <summary>
+        /// Tests that sending configuration to the DAB engine post-startup will properly hydrate even with data-source-files specified.
+        /// </summary>
+        [TestCategory(TestCategory.MSSQL)]
+        [TestMethod("Validates RuntimeConfig setup for post-configuraiton hydration with datasource-files specified.")]
+        [DataRow(CONFIGURATION_ENDPOINT)]
+        [DataRow(CONFIGURATION_ENDPOINT_V2)]
+        public async Task TestValidMultiSourceRunTimePostStartupConfigurations(string configurationEndpoint)
+        {
+            TestServer server = new(Program.CreateWebHostFromInMemoryUpdateableConfBuilder(Array.Empty<string>()));
+            HttpClient httpClient = server.CreateClient();
+
+            RuntimeConfig config = AuthorizationHelpers.InitRuntimeConfig(
+                entityName: POST_STARTUP_CONFIG_ENTITY,
+                entitySource: POST_STARTUP_CONFIG_ENTITY_SOURCE,
+                roleName: POST_STARTUP_CONFIG_ROLE,
+                operation: EntityActionOperation.Read,
+                includedCols: new HashSet<string>() { "*" });
+
+            // Set up Configuration with DataSource files.
+            config = config with { DataSourceFiles = new DataSourceFiles(new List<String>() { "file1", "file2" }) };
```

Comment:
If these files cannot be found, why does this not error out with File Not Found exception?

Reply:
From my understanding on the pr review call we had, for the static web app scenario We wanted the post startup to succeed in cases where the file specified was not found to maintain backward compatibility and continue to work for single db scenario. (Example: I think the example scenario mentioned was someone in their open source repo added these files, used same config in static web apps, forgot to remove data-source-files).